target/linux/ath79/base-files/etc/board.d/01_leds

```diff
@@ -25,6 +25,14 @@ case "$board" in
 	ucidef_set_led_usbdev "usb" "USB" "tp-link:green:usb" "1-1"
 	ucidef_set_led_wlan "wlan" "WLAN" "tp-link:green:wlan" "phy0tpt"
 	;;
+"tplink,tl-wr841n-v9")
+	ucidef_set_led_netdev "wan" "WAN" "tp-link:green:wan" "eth0"
+	ucidef_set_led_switch "lan1" "LAN1" "tp-link:green:lan1" "switch0" "0x10"
+	ucidef_set_led_switch "lan2" "LAN2" "tp-link:green:lan2" "switch0" "0x08"
+	ucidef_set_led_switch "lan3" "LAN3" "tp-link:green:lan3" "switch0" "0x04"
+	ucidef_set_led_switch "lan4" "LAN4" "tp-link:green:lan4" "switch0" "0x02"
+	ucidef_set_led_wlan "wlan" "WLAN" "tp-link:green:wlan" "phy0tpt"
```

Comment:
Shouldn't all the above go into the device-tree as well?
See the section on `linux,default-trigger` and `trigger-sources` in [Documentation/devicetree/bindings/leds/common.txt](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/leds/common.txt) of the Linux source tree.

Reply:
yes, I meant to do them, but I'm stuck at the wan port led thing

Reply:
as we still rely on the swconfig driver I guess we'll still need the userspace helpers for the switch-port leds, ie. LAN.
Logic suggests that wan should be doable using `trigger-sources`, but apparently this has only been implemented for `ledtrig-usbport`.
I reckon adding support for `trigger-sources` from device-tree into ledtrig-netdev and ledtrig-disk would warmy be welcomed upstream... The same would also be needed for our local `530-ath9k_extra_leds.patch` as well as `930-ath10k_add_tpt_led_trigger.patch`, but all in all, that's not too hard to do looking at how it works in [`ledtrig-usbport.c`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/usb/core/ledtrig-usbport.c#n151)...

Reply:
btw, do you know if AR934X built-in switch can trigger port5 like "switch0" "0x02" and the rest of them ? right now, wan is detected as port5 on the switch